packages/EasyErrorHandler/src/Bridge/Symfony/Providers/ErrorCodesByEnumProvider.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyErrorHandler\Bridge\Symfony\Providers;
+
+use EonX\EasyErrorHandler\Annotations\AsErrorCodes;
+use EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderInterface;
+use PhpParser\Error;
+use PhpParser\Node\Stmt\Enum_;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+use ReflectionClass;
+
+final class ErrorCodesByEnumProvider implements ErrorCodesProviderInterface
+{
+    private Parser $parser;
+
+    public function __construct(private string $projectDir)
+    {
+        $this->parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+    }
+
+    public function provide(): array
+    {
+        $enums = $this->locateErrorCodesEnums();
+
+        if (\count($enums) === 0) {
+            return [];
+        }
+
+        $errorCodes = [];
+        foreach ($enums as $enum) {
+            $cases = $enum::cases();
+            foreach ($cases as $case) {
+                $errorCodes[$case->name] = $case->value;
+            }
+        }
+
+        return $errorCodes;
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    private function locateErrorCodesEnums()
+    {
+        $directory = new \RecursiveDirectoryIterator($this->projectDir . '/src');
+        $iterator = new \RecursiveIteratorIterator($directory);
+        $regex = new \RegexIterator($iterator, '/Enum\/.*Enum\.php$/');
```

Comment:
What if there is another namespace structure in the app? For example, `Enums` folder instead of `Enum`. We have an exact criteria - the attribute, so why do we need to introduce the regex?

Reply:
We need it to avoid parsing all files in the project. It's prefilter. 

Maybe add a configuration parameter for this? And by default use `*.php`

Wdyt?

Reply:
Fixed